src/kernels/composable_kernel/asm/igemm_v4r1_dynamic.s

```diff
@@ -0,0 +1,6178 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+; generated by igemm_codegen.py
+;
+.macro .v_u32_div v_q, v_n, v_d, v_tmp4, s_tmp4
```

Comment:
Hi @atamazov , this whole assembly code is generated by [iGEMMgen](https://github.com/ROCmSoftwarePlatform/iGEMMgen). It is the exact same algorithm of igemm, used in current MIOpen, with only difference that it can compute index at runtime. So, if asking who can review, indeed I don't know who's best candidate. maybe @asroy ?

Reply:
@carlushuang I recommend either Ilya (@Kirpich30000) or Kamil (@shurale-nkn) plus whoever you would like to invite.

Reply:
There is no need to review the algorithm in detail, but there are aspects related to integrating asm code into MIOpen that I would like these guys to look at.

Reply:
Sure. Hi @Kirpich30000 @shurale-nkn, please help look at this .s file. since this file is generated by [iGEMMgen](https://github.com/ROCmSoftwarePlatform/iGEMMgen) so I need to modify that repo, if there's anything wrong or not proper, consider integration into MIOpen